Demo/Common/IntegrationExamples/AdvancedFlowExample.swift

```diff
@@ -0,0 +1,374 @@
+        if let paymentComponent = component as? PaymentComponent {
+            paymentComponent.delegate = delegate
+        }
+
+        if let actionComponent = component as? ActionComponent {
+            actionComponent.delegate = delegate
+        }
+
+        advancedFlowComponent = component
+        guard component.requiresModalPresentation else {
+            presenter?.present(viewController: component.viewController, completion: nil)
+            return
+        }
+
+        let navigation = UINavigationController(rootViewController: component.viewController)
+        component.viewController.navigationItem.rightBarButtonItem = .init(barButtonSystemItem: .cancel,
+                                                                           target: self,
+                                                                           action: #selector(cancelPressed))
+        presenter?.present(viewController: navigation, completion: nil)
+    }
+
+    @objc private func cancelPressed() {
+        advancedFlowComponent?.cancelIfNeeded()
+        presenter?.dismiss(completion: nil)
+    }
+
+}
+
+extension AdvancedFlowExample: StoredPaymentMethodsDelegate {
+    func disable(storedPaymentMethod: StoredPaymentMethod, completion: @escaping (Bool) -> Void) {
+        let request = DisableStoredPaymentMethodRequest(recurringDetailReference: storedPaymentMethod.identifier)
+        palApiClient.perform(request) { [weak self] result in
+            self?.handleDisableResult(result, completion: completion)
+        }
+    }
+
+    private func handleDisableResult(_ result: Result<DisableStoredPaymentMethodRequest.ResponseType, Error>, completion: (Bool) -> Void) {
+        switch result {
+        case let .failure(error):
+            presentAlert(with: error, retryHandler: nil)
+            completion(false)
+        case let .success(response):
+            completion(response.response == .detailsDisabled)
+        }
+    }
+}
+
+extension AdvancedFlowExample: PresentationDelegate {
+
+    internal func present(component: PresentableComponent) {
+        present(component, delegate: self)
+    }
+}
+
+extension AdvancedFlowExample: ActionComponentDelegate {
+
+    internal func didFail(with error: Error, from component: ActionComponent) {
+        finish(with: error)
+    }
+
+    internal func didComplete(from component: ActionComponent) {
+        finish(with: .received)
+    }
+
+    internal func didProvide(_ data: ActionComponentData, from component: ActionComponent) {
+        (component as? PresentableComponent)?.viewController.view.isUserInteractionEnabled = false
+        let request = PaymentDetailsRequest(
+            details: data.details,
+            paymentData: data.paymentData,
+            merchantAccount: ConfigurationConstants.current.merchantAccount
+        )
+        apiClient.perform(request, completionHandler: paymentResponseHandler)
```

Comment:
same as above.

Reply:
Made the change.da858bb9a858774fa139005a65bb215f94ed0c10